docs/pages/application-access/cloud-apis/aws-console.mdx

```diff
@@ -1,186 +1,395 @@
-</Admonition>
-
-### Create the example Read Only role
-
-Go to the IAM -> Access Management -> [Roles](https://console.aws.amazon.com/iamv2/home#/roles).
-Press Create Role.
 
-![Create Role Step 1](../../../img/application-access/create-role-example-readonly-1.png)
+- An AWS EC2 instance or Elastic Kubernetes Service (EKS) cluster where you will
+  run the Teleport Application Service. EC2 instances must be running a Linux
+  distribution. We recommend starting with a fresh demo instance or EKS cluster
+  to get familiar with the procedure before following this guide in production.
+
+- Permissions to create IAM roles and policies in the AWS account you want to
+  connect.
+
+- `aws` command line interface (CLI) tool in PATH. Read the AWS documentation to
+  [install or update the latest version of the AWS
+  CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-install.html).
+
+- If you plan to run the Teleport Application Service on EKS, you will need an
+  IAM OIDC provider running in your Kubernetes cluster. See the [AWS
+  documentation](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
+  for how to create an IAM OIDC provider.
+
+  To check whether you have an IAM OIDC provider running in your cluster, run
+  the following `aws` command, assigning <Var name="eks-region" /> to the
+  region where your EKS cluster is running and <Var name="cluster-name" /> to
+  the name of your Kubernetes cluster:
+
+  ```code
+  $ aws --region=<Var name="eks-region" /> eks describe-cluster --name <Var name="cluster-name"/> --query "cluster.identity.oidc.issuer" --output text
+  ```
+
+  If you have an IAM OIDC provider associated with your cluster, this command
+  will print its ID.
+
+## Step 1/4. Configure AWS IAM
+
+In this section, you will configure AWS IAM resources to allow the Teleport
+Application Service to proxy AWS APIs.
+
+The rest of the guide assumes that you are protecting access to an IAM role
+called `ExampleReadOnlyAccess`, which allows read-only access to AWS resources
+through the AWS-managed `ReadOnlyAccess` policy. We recommend following this
+guide in a demo environment with the `ExampleReadOnlyAccess` role, and
+protecting production AWS roles when you are familiar with the procedure.
+
+You will create the following resources:
+
+|Name | Resource | Function                                   |
+|---|----------|--------------------------------------------|
+|`ExampleReadOnlyAccess`| IAM role | Example role to protect access to with Teleport. |
+|`TeleportAWSAccess`|IAM role|Allows the Application Service to assume other roles in order to proxy user requests to AWS.|
+|`AssumeRole`|IAM policy|Allows the Application Service to assume other roles in order to proxy user requests to AWS.|
+|`TeleportAWSAccess` (for EC2 deployments) |EC2 instance profile|Associates the `TeleportAWSAccess` role with your EC2 instance.|
+
+### Configure a role to enable access to
```

Comment:
I've suggested "Configure a role for Teleport users to request", which I think makes more sense for this setup step. Otherwise, this section seems to overlap with the later section called, "Give the Teleport Application Service permissions to assume roles".